lms/djangoapps/discussion_api/api.py

```diff
@@ -276,7 +286,15 @@ def get_thread_list(request, course_key, page, page_size, topic_id_list=None, te
         "per_page": page_size,
         "text": text_search,
     }
+
     text_search_rewrite = None
+
+    if view:
+        if view in ["unread", "unanswered"]:
+            query_params[view] = "true"
+        else:
+            HttpResponseBadRequest("Invalid value for view: {}".format(view))
```

Comment:
I changed this to a Validation error.

The Http404 is an exception and not a response object. 
Http404 is used in various places in api.py. I would rather make this change in a different PR once I determine the best way to return these errors.